owncloudApp/src/main/res/layout/owncloud_toolbar.xml

```diff
@@ -0,0 +1,97 @@
+  -->
+<com.google.android.material.appbar.AppBarLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:id="@+id/owncloud_app_bar"
+    android:layout_width="match_parent"
+    android:layout_height="?android:actionBarSize"
+    android:theme="@style/ownCloud.Appbar">
+
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:id="@+id/root_toolbar"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent">
+
+        <com.google.android.material.imageview.ShapeableImageView
+            android:id="@+id/root_toolbar_left_icon"
+            android:layout_width="wrap_content"
+            android:layout_height="0dp"
+            android:layout_marginHorizontal="@dimen/standard_half_margin"
+            android:padding="@dimen/standard_half_padding"
+            android:src="@drawable/ic_drawer_icon"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent" />
+
+        <androidx.appcompat.widget.SearchView
+            android:id="@+id/root_toolbar_search_view"
+            android:layout_width="0dp"
+            android:layout_height="0dp"
+            android:gravity="center_vertical"
+            android:paddingHorizontal="@dimen/standard_half_padding"
+            android:visibility="gone"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toStartOf="@id/root_toolbar_avatar"
+            app:layout_constraintStart_toEndOf="@id/root_toolbar_left_icon"
+            app:layout_constraintTop_toTopOf="parent"
+            app:queryHint="@string/actionbar_search"
+            app:searchHintIcon="@null" />
+
+        <com.google.android.material.textview.MaterialTextView
+            android:id="@+id/root_toolbar_title"
+            android:layout_width="0dp"
+            android:layout_height="0dp"
+            android:layout_marginStart="@dimen/standard_margin"
+            android:layout_marginEnd="@dimen/standard_half_margin"
+            android:ellipsize="end"
+            android:gravity="center_vertical"
+            android:maxLines="1"
+            android:paddingHorizontal="@dimen/standard_half_padding"
+            android:textColor="@color/white"
+            android:textSize="@dimen/toolbar_title_text_size"
+            app:drawableEndCompat="@drawable/ic_search"
+            app:drawableTint="@color/white"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toStartOf="@id/root_toolbar_avatar"
+            app:layout_constraintStart_toEndOf="@id/root_toolbar_left_icon"
+            app:layout_constraintTop_toTopOf="parent"
+            tools:text="ownCloud" />
+
+        <com.google.android.material.imageview.ShapeableImageView
```

Comment:
Why `ShapeableImageView`? To be protected against design changes, right? 😄

Reply:
Nothing special, I have changed it to AppCompatImageView. We will move to material in the future 👍